src/lsm/lsm_cursor.c

```diff
@@ -921,19 +948,16 @@ __clsm_next(WT_CURSOR *cursor)
 		 * forward.
 		 */
 		if (F_ISSET(clsm, WT_CLSM_MULTIPLE)) {
-			check = false;
 			WT_FORALL_CURSORS(clsm, c, i) {
 				if (!F_ISSET(c, WT_CURSTD_KEY_INT))
 					continue;
-				if (check) {
+				if (c != clsm->current) {
```

Comment:
Again a simplification at the cost of extra key comparisons -- previously we skipped doing any comparisons on cursors until we went past the "current" cursor.  That made me worry about records appearing in the live chunk at read-uncommitted, hence this change.  I don't expect a noticeable perf impact from this change but it also isn't essential to fix the real issue.